generator/uscore/uscore_generator.rb

```diff
@@ -1122,34 +1122,37 @@ def create_search_validation(sequence)
           case type
           when 'Period', 'date'
             search_validators += %(
-                value_found = resolve_element_from_path(resource, '#{path_parts.join('.')}') { |date| validate_date_search(value, date) }
-                assert value_found.present?, '#{element} on resource does not match #{element} requested'
+                values_found = resolve_path(resource, '#{path_parts.join('.')}')
+                match_found = values_found.any? { |date| validate_date_search(value, date) }
+                assert match_found, "#{element} in #{sequence[:resource]}/\#{resource.id} (\#{values_found}) does not match #{element} requested (\#{value})"
```

Comment:
Does it make any sense to take some/all of this matching code and move it to a function? Or at least the assertion failure message, since it's the same for pretty much all of the searches listed here.

Reply:
Stylistically I think the generated code is what matters.  Does this comment apply to what is generated, or this code here?  I bring it up because I'd personally like to avoid the situation where this code is clean but then it generates very sparse tests that are just a bunch of function calls that only get called once in a given sequence (making it hard to follow).

Reply:
I can separate the assertion failure message or the validation code in the generator do remove some of the duplication. This won't affect the generated code.

Reply:
I updated it so that the type specific stuff is in its own function in the generator.